x-pack/plugins/osquery/server/_telemetry/sender.ts

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { CoreStart, ElasticsearchClient, Logger } from 'src/core/server';
+import type { TelemetryPluginStart, TelemetryPluginSetup } from 'src/plugins/telemetry/server';
+import { cloneDeep } from 'lodash';
+import axios from 'axios';
+import type { InfoResponse } from '@elastic/elasticsearch/lib/api/types';
+import { TelemetryQueue } from './queue';
+import type { OsqueryTelemetryChannel, OsqueryTelemetryChannelEvents } from './types';
+
+/**
+ * Simplified version of https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/server/lib/telemetry/sender.ts
```

Comment:
Thanks for the attribution. This will help Kibana generalise event sending instrumentation that can we can all use.
cc/ @afharo @Bamieh @lizozom